app/es2015/lib/app-start.js

```diff
@@ -18,57 +18,57 @@
+  const Able = require('lib/able');
+  const AppView = require('views/app');
+  const Assertion = require('lib/assertion');
+  const AuthErrors = require('lib/auth-errors');
+  const Backbone = require('backbone');
+  const BaseAuthenticationBroker = require('models/auth_brokers/base');
+  const CloseButtonView = require('views/close_button');
+  const ConfigLoader = require('lib/config-loader');
+  const Constants = require('lib/constants');
+  const Environment = require('lib/environment');
+  const ErrorUtils = require('lib/error-utils');
+  const FormPrefill = require('models/form-prefill');
+  const FxaClient = require('lib/fxa-client');
+  const FxDesktopV1AuthenticationBroker = require('models/auth_brokers/fx-desktop-v1');
+  const FxDesktopV2AuthenticationBroker = require('models/auth_brokers/fx-desktop-v2');
+  const FxDesktopV3AuthenticationBroker = require('models/auth_brokers/fx-desktop-v3');
+  const FxFennecV1AuthenticationBroker = require('models/auth_brokers/fx-fennec-v1');
+  const FxFirstrunV1AuthenticationBroker = require('models/auth_brokers/fx-firstrun-v1');
+  const FxFirstrunV2AuthenticationBroker = require('models/auth_brokers/fx-firstrun-v2');
+  const FxiOSV1AuthenticationBroker = require('models/auth_brokers/fx-ios-v1');
+  const FxiOSV2AuthenticationBroker = require('models/auth_brokers/fx-ios-v2');
+  const HeightObserver = require('lib/height-observer');
+  const IframeChannel = require('lib/channels/iframe');
+  const InterTabChannel = require('lib/channels/inter-tab');
+  const MarketingEmailClient = require('lib/marketing-email-client');
+  const Metrics = require('lib/metrics');
+  const Notifier = require('lib/channels/notifier');
+  const NullChannel = require('lib/channels/null');
+  const OAuthClient = require('lib/oauth-client');
+  const OAuthRelier = require('models/reliers/oauth');
+  const OriginCheck = require('lib/origin-check');
+  const p = require('lib/promise');
+  const ProfileClient = require('lib/profile-client');
+  const RedirectAuthenticationBroker = require('models/auth_brokers/redirect');
+  const RefreshObserver = require('models/refresh-observer');
+  const Relier = require('models/reliers/relier');
+  const Router = require('lib/router');
+  const SameBrowserVerificationModel = require('models/verification/same-browser');
+  const ScreenInfo = require('lib/screen-info');
+  const SentryMetrics = require('lib/sentry');
+  const Session = require('lib/session');
+  const Storage = require('lib/storage');
+  const StorageMetrics = require('lib/storage-metrics');
+  const SyncRelier = require('models/reliers/sync');
+  const Translator = require('lib/translator');
+  const UniqueUserId = require('models/unique-user-id');
+  const Url = require('lib/url');
+  const User = require('models/user');
+  const WebChannel = require('lib/channels/web');
+  const WebChannelAuthenticationBroker = require('models/auth_brokers/web-channel');
 
   function Start(options) {
     options = options || {};
```

Comment:
ES6 specific rules at http://eslint.org/docs/rules/#ecmascript-6

Not sure what rules we'd want to potentially add some of these:

```json
    "no-duplicate-imports": 2,
    "no-var": 0,
    "object-shorthand": 1,
    "prefer-arrow-callback": 1,
    "prefer-const": 1,
```

Running that seems to give me `✖ 2514 problems (0 errors, 2514 warnings)` (and none of them auto-fixable).

Possibly add some of these as well:

- `prefer-rest-params`: require rest parameters instead of arguments
- `prefer-spread`: require spread operators instead of .apply()
- `prefer-template`: require template literals instead of string concatenation
- `sort-imports`: enforce sorted import declarations within modules